services/auth/service/role_service.go

```diff
@@ -180,3 +180,21 @@ func GetUsersByRole(role models.Role) ([]string, error) {
 
 	return userids, nil
 }
+
+/*
+	Returns role stats
+*/
+func GetStats() (map[string]interface{}, error) {
```

Comment:
You can simplify and remove most of this code using the `db.GetStats` helper to generate the states of the `roles` field of the collection.